.github/workflows/tests.yml

```diff
@@ -384,6 +394,14 @@ jobs:
             xcrun llvm-cov export -format="lcov" build/*/src/*.so \
               -instr-profile default.profdata > info.lcov
           fi
+
+      - name: Upload test results to Codecov
+        if: ${{ !cancelled() }}  # Run even if tests fail
+        uses: codecov/test-results-action@v1
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          os: ${{ matrix.codecov-test-results-action-os }}
```

Comment:
Is there a reason we need to manually specify the os - the docs say "Override the assumed OS.", so I would have assumed that the os can be autodetected somehow? I was thinking of saving having to hard code the os as a variable for every element of the test matrix.